lib/rage/controller/api.rb

```diff
@@ -283,4 +285,33 @@ def headers
     @__headers = {}.merge!(@__headers) if DEFAULT_HEADERS.equal?(@__headers)
     @__headers
   end
+
+  class Request
```

Comment:
Unfortunately, it's a bit more complicated.

We can't iterate through the whole env normalizing header names, as it is inefficient. A user might only need one header value, but your implementation will loop through all of them to do that.

On top of that, the `Content-Type` and `Content-Length` headers are not getting the `HTTP_` prefix.

Instead, nothing should be done until an actual header is requested. When it is, the code should look something like this:

```ruby
if requested_header.start_with?("HTTP_")
  env[requested_header]
else
  normalized_name = ... # normalize the header name, e.g. User-Agent -> HTTP_USER_AGENT
  env[normalized_name]
end
```

Also, the code should somehow make exceptions for the two Content headers.